src/one/nio/serial/DataStream.java

```diff
@@ -183,7 +183,8 @@ public void writeFrom(long address, int len) throws IOException {
     }
 
     public int read() throws IOException {
-        return unsafe.getByte(array, alloc(1));
+        long offset = alloc(1);
```

Comment:
Description of the pull request says:

> Some DataStream methods ignore changes the array variable. This pull request fixes that.

How does it fix that? All the changes in `DataStream` class are semantically equivalent and just increase the code base by introducing a new local variable.

Reply:
All write methods use new local variable, for example:
https://github.com/odnoklassniki/one-nio/blob/4bd740b6619c5bc7dd85d428df99926239e4b7aa/src/one/nio/serial/DataStream.java#L73-L76
So the `PersistStream.alloc()` method can assign a new array to an `array` field (line 76):
https://github.com/odnoklassniki/one-nio/blob/4bd740b6619c5bc7dd85d428df99926239e4b7aa/src/one/nio/serial/PersistStream.java#L72-L79
before the `unsafe.putByte()` method receives an array to write to.
If the `DataStream.write()` method was written in one line, the unsafe.putByte() method would receive the old array and PersistStream class would not work correctly. So the changes are not semantically equivalent.
I use a class similar to PersistStream, but not for writing, but for reading. Because some read methods are written in one line, I cannot replace the `array` field in the `alloc()` method. I have to override all this methods.